### ISSUE TITLE
resolve `.*` dependencies to array of keys/indexes

### ISSUE THREAD

**Issue by Guria** [cerebral/cerebral#636]
I have played a bit with `.*` optimizations and applyed them on todomvc. https://github.com/cerebral/cerebral/pull/634/files
`.*` - describes interest only in keys/indexes changes, but not on actual nested objects.
Along with `get(tag)` function available in computed it allows to show interest in exact sub objects and exact keys on it:
```js
import {compute} from 'cerebral'
import {state} from 'cerebral/tags'

export default compute(
  state`app.todos.*`,
  state`app.filter`,
  (todos, filter, get) => {
    return Object.keys(todos).filter((ref) => {
      const completed = get(state`app.todos.${ref}.completed`)

      return (
        filter === 'all' ||
        (filter === 'completed' && completed) ||
        (filter === 'active' && !completed)
      )
    })
  }
```

But since `.*`resolves to full object/array inattentive developer could forget to use get and access sub objects and their properties directly:

```js
import {compute} from 'cerebral'
import {state} from 'cerebral/tags'

export default compute(
  state`app.todos.*`,
  state`app.filter`,
  (todos, filter) => {
    return Object.keys(todos).filter((ref) => {
      const completed = todos[ref].completed

      return (
        filter === 'all' ||
        (filter === 'completed' && completed) ||
        (filter === 'active' && !completed)
      )
    })
  }
```
And it will keep interest on todos.*.completed untracked, so it wouldn't recalculate on `completed` property change.

So here is my suggestion. We should resolve `.*` dependencies to only array of keys/indexes of original object/array. Therefore developer would be forced to use `get` and dependencies would be tracke correctly:
```js
import {compute} from 'cerebral'
import {state} from 'cerebral/tags'

export default compute(
  state`app.todos.*`,
  state`app.filter`,
  (todoKeys, filter, get) => {
    return todoKeys.filter((ref) => {
      const completed = get(state`app.todos.${ref}.completed`)

      return (
        filter === 'all' ||
        (filter === 'completed' && completed) ||
        (filter === 'active' && !completed)
      )
    })
  }
```

As additional optimization we should warn when `.*` is applied to a path that contains neither Object nor Array.

**Comment by gaspard** [comment]
Very good idea 💡!

**Comment by Guria** [comment]
landed in test via #639
